run.sh

```diff
@@ -1,152 +1,318 @@
+    local pytest_args=()
+    local pytest_numprocesses='auto'
+
+    ## handle environment variables
+
+    # split TEST_LIBRARY on ','
+    IFS=',' read -r -a libraries <<< "${TEST_LIBRARY:-}"
+
+    ## parse command arguments
+
+    # parse flags
+    while [[ "$#" -gt 0 ]]; do
+        case "$1" in
+            +h|++help)
+                help
+                exit
+                ;;
+            +d|++docker)
+                docker=1
+                ;;
+            +G|++scenario-group)
+                if [[ "$#" -eq 1 ]]; then
+                  error "missing argument value for: $1"
+                  help
+                  exit 64
+                fi
+                # upcase via ${2^^} is unsupported on bash 3.x
+                mapfile -t group <<< "$(lookup_scenario_group "$(echo "$2" | upcase)")"
+                scenarios+=("${group[@]}")
+                shift
+                ;;
+            +S|++scenario|-S|--scenario)
+                # this also catches '-S' even though it's a pytest flag because
+                # there may be special treatment for specific scenarios
+                if [[ "$#" -eq 1 ]]; then
+                  error "missing argument value for: $1"
+                  hint
+                  exit 64
+                fi
+                # upcase via ${2^^} is unsupported on bash 3.x
+                scenarios+=("$(echo "$2" | upcase)")
+                shift
+                ;;
+            +l|++library)
+                if [[ "$#" -eq 1 ]]; then
+                  error "missing argument value for: $1"
+                  hint
+                  exit 64
+                fi
+                libraries+=("$2")
+                shift
+                ;;
+            ++)
+                # ignore and stop flag processing to force remainder to be captured as is
+                shift
+                break
+                ;;
+            +*)
+                # unknown flag: be helpful
+                error "unknown flag: $1"
+                hint
+                exit 64
+                ;;
+            *)
+                # handle positional arguments
+                if [[ "$1" =~ [A-Z0-9_]+_SCENARIOS$ ]]; then
+                    mapfile -t group <<< "$(lookup_scenario_group "$1")"
+                    scenarios+=("${group[@]}")
+                elif [[ "$1" =~ ^[A-Z0-9_]+$ ]]; then
+                    scenarios+=("$1")
+                else
+                    # pass any unmatched arguments to pytest
+                    pytest_args+=("$1")
+                fi
+                ;;
+        esac
+        shift
+    done
+
+    # capture remainder of arguments to pass as-is for pytest
+    pytest_args+=("$@")
+
+    ## prepare commands
+
+    # when no scenario is provided, use a nice default
+    if [[ "${#scenarios[@]}" -lt 1 ]]; then
+        scenarios+=('DEFAULT')
     fi
 
-    # TODO: default to "auto" when dotnet is fixed
-    PYTEST_WORKER_COUNT=${PYTEST_WORKER_COUNT:-$DEFAULT_COUNT}
```

Comment:
How is it possible that this environment variable has been removed without any communication or _replacement_? Just stop messing with things that are used in other teams CI setups! @lloeki @cbeauchesne